python-checks/src/main/java/org/sonar/python/checks/tests/DedicatedAssertionCheck.java

```diff
@@ -55,30 +56,33 @@ public class DedicatedAssertionCheck extends PythonSubscriptionCheck {
   private static final String IS = "#is#";
   private static final String IS_NOT = "#is-not#";
   private static final String IN = "#in#";
+  private static final String NOT_IN = "#not-in#";
 
   private static final String FIRST = "first";
   private static final String SECOND = "second";
 
   private static final String ROUND = "round";
   private static final String IS_INSTANCE = "isinstance";
 
-  private static final Map<String, String> ASSERT_TRUE_SUGGESTIONS = Map.of(
-    "==", ASSERT_EQUAL,
-    "!=", ASSERT_NOT_EQUAL,
-    ">", "assertGreater",
-    ">=", "assertGreaterEqual",
-    "<", "assertLess",
-    "<=", "assertLessEqual",
-    IN, "assertIn",
-    IS, "assertIs",
-    IS_NOT, "assertIsNot",
-    IS_INSTANCE, "assertIsInstance"
+  private static final Map<String, String> ASSERT_TRUE_SUGGESTIONS = Map.ofEntries(
+    Map.entry("==", ASSERT_EQUAL),
+    Map.entry("!=", ASSERT_NOT_EQUAL),
+    Map.entry(">", "assertGreater"),
+    Map.entry(">=", "assertGreaterEqual"),
+    Map.entry("<", "assertLess"),
+    Map.entry("<=", "assertLessEqual"),
+    Map.entry(IN, "assertIn"),
+    Map.entry(NOT_IN, "assertNotIn"),
+    Map.entry(IS, "assertIs"),
+    Map.entry(IS_NOT, "assertIsNot"),
+    Map.entry(IS_INSTANCE, "assertIsInstance")
   );
```

Comment:
Why did you replace `Map.of` with `Map.ofEntries`? IMO it reduces the readability of the code.

Reply:
I fully agree with you, but we don't have the choice :D

I was thinking that `Map.of` was using a **varargs** style parameter which would allow an unlimited number of arguments, but that's not the case.
Instead, this methods is overridden from 0 to 10 different map values, meaning that if you want more that 10 values in your `Map`, you cannot use it and must switch to an alternative.

![image](https://user-images.githubusercontent.com/110470341/187904395-b60caff7-b282-45ea-8d72-d066588ebe74.png)

In that case we are unlucky : with the **NOT_IN** map value we add, we now have 11 values.
I searched for alternatives and apparently the only one available is the usage of `Map.ofEntries`, which is using **varargs**.

Reply:
Ah, you are right. I totally forgot that this method is limited. So everything is fine.